test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+        },
+        -- %H
+        {
+            buf = '00',
+            fmt = '%H',
+        },
+        {
+            buf = '23',
+            fmt = '%H',
+        },
+        -- %I
+        {
+            buf = '01',
+            fmt = '%I',
+        },
+        {
+            buf = '12',
+            fmt = '%I',
+        },
+        -- %j
+        {
+            buf = '001',
+            fmt = '%j',
+        },
+        {
+            buf = '31-12-2024 366',
+            fmt = '%d-%m-%Y %j',
+        },
+        -- %k
+        {
+            buf = ' 0',
+            fmt = '%k',
+        },
+        {
+            buf = '23',
+            fmt = '%k',
+        },
+        -- %l
+        {
+            buf = ' 1',
+            fmt = '%l',
+        },
+        {
+            buf = '12',
+            fmt = '%l',
+        },
+        -- %m
+        {
+            buf = '01',
+            fmt = '%m',
+        },
+        {
+            buf = '12',
+            fmt = '%m',
+        },
+        -- %M
+        {
+            buf = '00',
+            fmt = '%M',
+        },
+        {
+            buf = '59',
+            fmt = '%M',
+        },
+        -- %n
+        {
+            buf = '\n',
+            fmt = '%n',
+        },
+        -- %O, modifier, skipped.
+        -- %p
+        {
+            buf = 'PM',
+            fmt = '%p',
+        },
+        {
+            buf = 'AM',
+            fmt = '%p',
+        },
+        -- %r, same as "%I:%M:%S %p".
+        {
+            buf = '12:32:01 AM',
+            fmt = '%r',
+        },
+        {
+            buf = '12:32:01 PM',
+            fmt = '%r',
+        },
+        -- %R, same as "%H:%M".
+        {
+            buf = '18:20',
+            fmt = '%R',
+        },
+        -- %s
+        {
+            buf = '26-08-2024 1724630400',
+            fmt = '%d-%m-%Y %s',
+        },
+        -- %S
+        {
+            buf = '00',
+            fmt = '%S',
+        },
+        {
+            buf = '59',
+            fmt = '%S',
+        },
+        -- %t
+        {
+            buf = '26-08-2024 \t',
+            fmt = '%d-%m-%Y %t',
+        },
+        -- %T, same as "%H:%M:%S".
+        {
+            buf = '00:00:00',
+            fmt = '%T',
+        },
+        {
+            buf = '23:59:59',
+            fmt = '%T',
+        },
+        -- %u
+        -- Works for 4 ('Thursday') only when date is not
+        -- complete.
+        {
+            buf = '4',
+            fmt = '%u',
+        },
+        {
+            buf = '30-08-2024 5',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '26-08-2024 1',
+            fmt = '%d-%m-%Y %u',
+        },
+        -- %U
+        {
+            buf = '01-01-2024 00',
+            fmt = '%d-%m-%Y %U',
+        },
+        {
+            buf = '31-12-2024 52', -- "00 to 53"?
+            fmt = '%d-%m-%Y %U',
+        },
+        -- %v, same as "%e-%b-%".
+        {
+            buf = ' 1-Jan-1970',
```

Comment:
Do we need the leading whitespace? If yes, please add the corresponding comment.

Reply:
Added for every conversion specification.